.github/workflows/Bonsai.yml

```diff
@@ -0,0 +1,460 @@
+  # Build, test, and package
+  # =====================================================================================================================================================================
+  build-and-test:
+    needs: create-build-matrix
+    strategy:
+      fail-fast: false
+      matrix: ${{fromJSON(needs.create-build-matrix.outputs.matrix)}}
+    name: ${{matrix.platform.name}} ${{matrix.configuration}}
+    runs-on: ${{matrix.platform.os}}
+    env:
+      IsReferenceDummyBuild: ${{matrix.dummy-build}}
+      UseRepackForBootstrapperPackage: ${{matrix.collect-packages && !matrix.dummy-build}}
+    steps:
+      # ----------------------------------------------------------------------- Checkout
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          ref: ${{matrix.checkout-ref}}
+
+      # ----------------------------------------------------------------------- Setup tools
+      - name: Setup .NET
+        uses: actions/setup-dotnet@v4
+        with:
+          dotnet-version: 8.x
+
+      - name: Setup Python 3.11
+        uses: actions/setup-python@v5
+        with:
+          python-version: '3.11'
+
+      # Legacy .NET command line tools are only used for building the installer
+      - name: Setup MSBuild command line
+        if: matrix.create-installer
+        uses: microsoft/setup-msbuild@v2
+      - name: Setup NuGet command line
+        if: matrix.create-installer
+        uses: NuGet/setup-nuget@v2
+        with:
+          nuget-version: 6.x
+
+      # ----------------------------------------------------------------------- Configure build
+      - name: Configure build
+        run: python .github/workflows/configure-build.py
+        env:
+          github_event_name: ${{github.event_name}}
+          github_ref: ${{github.ref}}
+          github_run_number: ${{github.run_number}}
+          release_is_prerelease: ${{github.event.release.prerelease}}
+          release_version: ${{github.event.release.tag_name}}
+          workflow_dispatch_version: ${{github.event.inputs.version}}
+          workflow_dispatch_will_publish_packages: ${{github.event.inputs.will_publish_packages}}
+   
+      # ----------------------------------------------------------------------- Build
+      - name: Restore
+        run: dotnet restore Bonsai.sln
+
+      - name: Build
+        run: dotnet build Bonsai.sln --no-restore --configuration ${{matrix.configuration}}
+
+      # ----------------------------------------------------------------------- Repack bootstrapper
+      # This happens before pack since the bootstrapper package uses it
+      - name: Repack bootstrapper
+        if: matrix.create-installer || env.UseRepackForBootstrapperPackage == 'true'
+        run: dotnet build Bonsai --no-restore --configuration ${{matrix.configuration}} -t:Repack
```

Comment:
Correct. The only reason the Bonsai.sln is repeated everywhere is that `dotnet` sees the `Bonsai.Setup.sln` and doesn't want to assume which solution you care about. (Ideally we'd just omit the solution from all commands otherwise.)